src/level/robotmain.cpp

```diff
@@ -3784,6 +3785,12 @@ void CRobotMain::CreateScene(bool soluce, bool fixScene, bool resetObject)
             m_build |= BUILD_FLAG;
         }
 
+        if (m_levelCategory == LevelCategory::GamePlus && !m_ui->GetPlusResearch() && !resetObject)  // new game plus?
```

Comment:
Ugly comment! Move the condition into a method or free function like "bool isNewGamePlus()".